app/src/main/kotlin/com/github/gotify/api/ClientFactory.kt

```diff
@@ -7,57 +7,78 @@ import com.github.gotify.client.api.UserApi
 import com.github.gotify.client.api.VersionApi
 import com.github.gotify.client.auth.ApiKeyAuth
 import com.github.gotify.client.auth.HttpBasicAuth
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import org.tinylog.kotlin.Logger
 
 internal object ClientFactory {
-    private fun unauthorized(baseUrl: String, sslSettings: SSLSettings): ApiClient {
-        return defaultClient(arrayOf(), "$baseUrl/", sslSettings)
+    private fun unauthorized(
+        settings: Settings,
+        sslSettings: SSLSettings,
+        baseUrl: String
+    ): ApiClient {
+        return defaultClient(arrayOf(), settings, sslSettings, baseUrl)
     }
 
     fun basicAuth(
-        baseUrl: String,
+        settings: Settings,
         sslSettings: SSLSettings,
         username: String,
         password: String
     ): ApiClient {
-        val client = defaultClient(
-            arrayOf("basicAuth"),
-            "$baseUrl/",
-            sslSettings
-        )
+        val client = defaultClient(arrayOf("basicAuth"), settings, sslSettings)
         val auth = client.apiAuthorizations["basicAuth"] as HttpBasicAuth
         auth.username = username
         auth.password = password
         return client
     }
 
-    fun clientToken(baseUrl: String, sslSettings: SSLSettings, token: String?): ApiClient {
-        val client = defaultClient(
-            arrayOf("clientTokenHeader"),
-            "$baseUrl/",
-            sslSettings
-        )
+    fun clientToken(settings: Settings, token: String? = settings.token): ApiClient {
+        val client = defaultClient(arrayOf("clientTokenHeader"), settings)
         val tokenAuth = client.apiAuthorizations["clientTokenHeader"] as ApiKeyAuth
         tokenAuth.apiKey = token
         return client
     }
 
-    fun versionApi(baseUrl: String, sslSettings: SSLSettings): VersionApi {
-        return unauthorized(baseUrl, sslSettings).createService(VersionApi::class.java)
+    fun versionApi(
+        settings: Settings,
+        sslSettings: SSLSettings = settings.sslSettings(),
+        baseUrl: String = settings.url
```

Comment:
Nice improvement (:.

Reply:
Yeah, I had to add the settings object to all client factory methods.  
That's was the motivation to simplify and unify the method signatures overall.